src/lib/components/workflow/parent-workflow-table.svelte

```diff
@@ -37,7 +37,7 @@
       <Link
         newTab
         href={routeForEventHistory({
-          namespace,
+          namespace: parentNamespaceName ?? namespace,
```

Comment:
Related, I think we need a `namespace: child.namespace` here ⬇️ 
https://github.com/temporalio/ui/blob/e781ade6f23344fbaa1260e5e5503e1d1fc07546/src/lib/components/workflow/child-workflows-table.svelte#L73